clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -280,7 +295,27 @@ void Sema::checkSYCLDeviceVarDecl(VarDecl *Var) {
   SourceRange Loc = Var->getLocation();
   llvm::DenseSet<QualType> Visited;
 
-  checkSYCLVarType(*this, Ty, Loc, Visited);
+  checkSYCLVarType(*this, Ty, Loc, Visited, false);
+}
+
+static void checkAccessorVarType(Sema &S, QualType Ty, SourceRange Loc) {
```

Comment:
Should it be a member of `SyclKernelFieldChecker` class?

Reply:
Good question.  Do you have a preference?

Reply:
I think it should too.

Reply:
done